crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -111,19 +127,84 @@ impl SettlementEncoder {
             .token_index(order.order_creation.buy_token)
             .expect("missing buy token with price");
 
+        let order_trade = OrderTrade {
+            trade: Trade {
+                order,
+                sell_token_index,
+                executed_amount,
+                scaled_fee_amount,
+            },
+            buy_token_index,
+        };
+        let execution = order_trade
+            .trade
+            .executed_amounts(*sell_price, *buy_price)
+            .context("impossible trade execution")?;
+
+        self.order_trades.push(order_trade);
+        Ok(execution)
+    }
+
+    // Fails if any used sell-token doesn't have a price.
+    pub fn add_liquidity_order_trade(
+        &mut self,
+        order: Order,
+        executed_amount: U256,
+        scaled_fee_amount: U256,
+    ) -> Result<TradeExecution> {
+        let sell_price = self
+            .clearing_prices
+            .get(&order.order_creation.sell_token)
+            .context("settlement missing sell token")?;
+
+        // Note, for the encoding strategy of liquidity orders, there
```

Comment:
@marcovc Is this okay for you?

The reason why I prefer to have it provided by the solvers is the following:
- If solvers do not provide the sell price for the liquidity orders, then the driver can choose it price freely up to a scalar: E.g. [sell_price, buy_price] or [a*sell_price, a*buy_price] would both be okay. In order to have it more deterministic and in order to give the solvers more control over the submitted solution, I would prefer to require the solvers to provide the sell_token_price for liquidity orders.

Reply:
Currently the json includes the order executed buy and sell amounts, I guess everything can be derived from that, right?

Reply:
Yes, we could just use the executed_amounts as sell and buy prices, thought this might be confusing: 
- If the sellTokenPrice is provided then we want to reuse it for liquidity order trades
- If the sellTokenPrice is not provided, then we would use the executed_buy_amount.

Hence, my proposal is to always provide the sellTokenPrice. But yeah, we can also go with  your logic. I don't have a strong opinion

Reply:
I think I might need to read your PR again, but I'm currently very busy. Please go ahead with your suggestion, I'm sure it makes sense and that we can adapt how we express that exchange rate later.

Reply:
:) I will wait for you @marcovc. If we merge it as it is and the solver does not provide the sell_token_price, then the solution will not be accepted. Feel free to huddle me, to speed it up. No need to read the whole PR.

Reply:
Marco convinced me to use the uniform clearing price vector only for price information of usual user orders. I will refactor this part.